ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java

```diff
@@ -114,7 +114,7 @@ private void createOrResumeXidTransaction(final XidTransactionID xidTransactionI
         } else {
             // begin a new tx and add it to the tx repository
             // TODO: Fix the tx timeout to accept a value from the client (WFLY-2789)
-            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.MAX_VALUE);
+            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.parseInt(System.getProperty("remote-tx-timeout", "31536000")));
```

Comment:
Well, lets first have optimal solution and than discuss backporting :)
System property feels like workaround for easy configuration enabled fix, but that doesn't always make it right.
Even if you look at comment above that line, we are already missing timeout propagation from client WFLY-2789, why not fix all this for good and be done with it?